src/LayerPlan.cpp

```diff
@@ -135,121 +135,92 @@ ExtruderTrain* LayerPlan::getLastPlannedExtruderTrain()
     return last_planned_extruder;
 }
 
-
-Polygons LayerPlan::computeCombBoundaryInside(const size_t max_inset)
+bool LayerPlan::CombBoundaryRequired()
 {
     const CombingMode combing_mode = Application::getInstance().current_slice->scene.current_mesh_group->settings.get<CombingMode>("retraction_combing");
-    if (combing_mode == CombingMode::OFF)
+    if (combing_mode == CombingMode::OFF || (layer_nr < 0 && combing_mode == CombingMode::NO_SKIN))
     {
-        return Polygons();
+        return false;
     }
-    if (layer_nr < 0)
-    { // when a raft is present
-        if (combing_mode == CombingMode::NO_SKIN)
+    return true;
+}
+
+Polygons LayerPlan::computeMinimumCombBoundary()
+{
+    Polygons comb_boundary;
+    if (CombBoundaryRequired())
+    {
+        if (layer_nr < 0)
         {
-            return Polygons();
+            comb_boundary = storage.raftOutline.offset(MM2INT(0.1));
         }
         else
         {
-            return storage.raftOutline.offset(MM2INT(0.1));
+            for (const SliceMeshStorage& mesh : storage.meshes)
+            {
+                const SliceLayer& layer = mesh.layers[static_cast<size_t>(layer_nr)];
+                if (mesh.settings.get<bool>("infill_mesh")) // don't process infill_mesh
+                {
+                    continue;
+                }
+                const coord_t offset = -10 - mesh.settings.get<coord_t>("wall_line_width_0");
+                for (const SliceLayerPart& part : layer.parts)
+                {
+                    comb_boundary.add(part.outline.offset(offset));
```

Comment:
Yes that was the old "correct" behaviour. @konskarm raised the same question but we both thought this implementation might be in improvement. That being said, this will be technically be a change in the behaviour so I will change the code per your suggestions.

Reply:
Well the problem is that it then doesn't effectively work any more as `NO_SKIN` will still go through skin at the top and bottom of the model and such.